chain/types/ethtypes/rlp.go

```diff
@@ -0,0 +1,180 @@
+package ethtypes
+
+import (
+	"bytes"
+	"encoding/binary"
+	"fmt"
+
+	"golang.org/x/xerrors"
+)
+
+// set a limit to make sure it doesn't loop infinitely
+const maxListElements = 500
```

Comment:
How was this number selected?

Reply:
I believe this is an arbitrary backstop number. Given that we use the RLP encoder/decoder for transactions only, we could lower this to something a lot more conservative, like 16 or 32.

Reply:
Made is stricter: 12.